### PR TITLE
Automatically build and push the docker image when tagging a new version

### LINKED ISSUE #1246: Use the release CI to build and publish the docker image
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

Currently, we expect to release the docker image when calling a new vote process. It's a bit trivial to request the docker image registry permission for all release managers.

### Solution

Automatically build and push the docker image if the new tag was created.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #1246

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    steps:
+
+      - name: Checkout Code Base
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 64
+
+      - name: Set ENV
+        run: |
+          echo "RELEASE_TAG=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
```
Comment (PragmaTwice) [inline .github/workflows/release.yaml]:
```suggestion
          echo "RELEASE_TAG=${GITHUB_REF#refs/*/v}" >> $GITHUB_ENV
```

Comment (PragmaTwice) [reply .github/workflows/release.yaml]: We can remove the `v` prefix.

**Review by PragmaTwice — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks all, merging...

@caipengbo Can help to push a new tag to have a try, the DOCKER_USERNAME and DOCKER_PASSWORD had existed before if my memory serves.

**Review by tisonkun — COMMENTED** [review]
This workflow can break Apache release process that any official release should go through a vote process.

What if any commit push a new tag and "release" a docker image that doesn't associate with a vote?

**Comment by tisonkun** [comment]
This is already discussed at:

* https://github.com/apache/incubator-kvrocks/pull/588#issuecomment-1131320266
* https://github.com/apache/incubator-kvrocks/pull/752#issuecomment-1194936068

**Comment by git-hulk** [comment]
OK, let's reverse this change and do them manually.

**Pull request merged**